.github/workflows/isotovideo-action.yml

```
@@ -9,11 +9,9 @@ jobs:
       image: "registry.opensuse.org/devel/openqa/containers/isotovideo:qemu-x86-jq"
     steps:
       - uses: actions/checkout@v2
-      - name: install jq
-        run: zypper -n in jq
 
       - name: Run isotovideo against test code
-        run: isotovideo qemu_no_kvm=1 casedir=.
+        run: isotovideo qemu_no_kvm=1 casedir=$(pwd)
 
       - name: fail if any test module failed
-        run: jq .result testresults/result-*.json | grep -v ok && echo "Test modules failed" && exit 1
+        run: jq .result testresults/result-*.json | grep ok || (echo "Test modules failed" && exit 1)
```

that `grep ok` would just mean that there is at least one test module that failed. But the purpose was to fail if any module fails. As we expect the execution in the example to fail what we can do is

```suggestion
        run: jq .result testresults/result-*.json | grep -v ok && (echo "At least one test module failed as expected" && exit 0)
```

---

i applied suggestion from @baierjan. review and resolve please

---

Yes, I saw that you applied the suggestion from @baierjan. The problem is that this way we are ignoring results either way. But we should check for the expected result which is actually that test modules *should* fail which serves as the example

---

Do you suggest that it is expected that the test will fail (i.e. the result of the isotovideo is no ok)? Running the pipeline locally yields `testresults/result-boot.json` with the `"result" : "ok"`, so the original code https://github.com/os-autoinst/os-autoinst-distri-example/blob/10683a4777aae56837fd0aaa789666914c5b71df/.github/workflows/isotovideo-action.yml#L19
only fails the pipeline because of the empty match from grep (i.e. after removing all ok results, there is no other record).

---

hm, you are right. That changed in a pull request from @Martchus some months ago. Then it should be

```
run: jq .result testresults/result-*.json | grep -v ok || (echo "Test modules failed" && exit 1)
```

right?

---

I believe not. The success of `grep -v ok` means something else than ok was matched (and that should be a failure); on the other hand, if `grep -v ok` fails (with exit code 1, btw.) it means there is no other result than ok, so it is actually a success. 
So what we want to have is
```sh
if $(jq .result testresults/result-*.json | grep -v ok); then echo "Test modules failed"; exit 1; else exit 0; fi
```
My original suggestion `jq .result testresults/result-*.json | grep -v ok && { echo "Test modules failed" && exit 1; } || true` should be the same. The "only" issue here is, that it masks error in jq. If we want to solve that, we need to introduce a little more logic.

Maybe
```sh
test $(jq .result testresults/result-*.json | grep -v ok | wc -l) -eq 0 || { echo "Test modules failed"; exit 1; }
```
could be more readable?